### PR TITLE
PRESIDECMS-2713 use faster Lucee 6 dbinfo when available

### PR DESCRIPTION
**Pull request by zspitzer**
https://presidecms.atlassian.net/browse/PRESIDECMS-2713

### CODE REVIEW

File: tests/server.json [hunk 1]
```diff
@@ -8,6 +8,9 @@
             "port":"9999"
         }
     },
+    "app":{
```
Comment (DominicWatson) [inline tests/server.json]: While we should probably be using a matrix of versions for support, we shouldn't be making a 6 snapshot our main Preside test platform.

We should ensure this continues to work on 5 as a priority over running the main test suite in 6.

**Review by DominicWatson — CHANGES_REQUESTED** [review]
Awesome looks great. Just one comment re changing the test suite Lucee version (we shouldn't be).

File: tests/server.json (hunk 1 above)
Comment (zspitzer) [reply tests/server.json]: sure, no problem, just wanted to see if it works with 6, was literally going back to mark this as a draft PR!

File: system/services/presideObjects/SqlSchemaSynchronizer.cfc [hunk 2]
```diff
@@ -188,7 +188,7 @@ component {
 
 		for( colName in ListToArray( arguments.dbFieldList ) ) {
 			colMeta = arguments.properties[ colName ];
-			if( listFindNoCase( tableColumns, colName ) && _skipSync( colMeta ) ) {
+			if( _skipSync( colMeta ) && ArrayContainsNoCase( tableColumns, colName ) ) {
```
Comment (zspitzer) [inline system/services/presideObjects/SqlSchemaSynchronizer.cfc:192]: did this boolean skip check first as it's cheaper?

File: system/services/presideObjects/SqlSchemaSynchronizer.cfc [hunk 3]
```diff
@@ -702,7 +702,29 @@ component {
 		return variables._tableCache[ arguments.dsn ][ arguments.tableName ] ?: false;
 	}
 
-	private query function _getTableColumns() {
+	private array function _getTableColumns() {
+		try {
+			if ( _getDbInfoService().hasModernDbInfo() ){
+				// with the faster version, we can fetch the entire schema in one call.
+				if ( !StructKeyExists( variables, "columnCache" )){
+					variables.columnCache =  _getTableColumnCache( argumentCollection = arguments );
```
Comment (zspitzer) [inline system/services/presideObjects/SqlSchemaSynchronizer.cfc:711]: it's my understanding this only runs once at startup? but it probably makes sense to purge this cache after the sync finishes, no need for a query with all the columns in the db to be left hanging in memory

**Comment by zspitzer** [comment]
Been testing, this reduces the db sync time by about roughly 50% on my fast laptop, 12s to 7s

**Comment by DominicWatson** [comment]
Great, thanks for the tidy up also. Will get this merged and in use internally for QA against existing Lucee.

**Pull request merged**